src/components/profile/WithdrawalModal.js

```diff
@@ -324,10 +332,162 @@ const StripeSetup = observer(({Cancel, Close}) => {
   );
 });
 
+// Circle, Step 2 - Link account
+const CircleSetup = observer(({Continue, Cancel}) => {
+  const [accountLoaded, setAccountLoaded] = useState(false);
+  const [errorMessage, setErrorMessage] = useState(undefined);
+  const [address, setAddress] = useState("");
+  const [chain, setChain] = useState("");
+  const [connected, setConnected] = useState(false);
+
+  useEffect(() => {
+    cryptoStore.LoadConnectedAccounts()
+      .then(() => {
+        setConnected(cryptoStore.CircleConnected());
+        setAccountLoaded(true);
+      });
+  }, []);
+
+  if(!accountLoaded) {
+    return (
+      <div className="withdrawal-confirmation">
+        <div className="withdrawal-confirmation__header">
+          { rootStore.l10n.withdrawal.set_up_withdrawal_circle }
+        </div>
+        <div className="withdrawal-confirmation__content">
+          <Loader />
+        </div>
+      </div>
+    );
+  }
+
+  if(!connected) {
+    return (
+      <div className="withdrawal-confirmation">
+        <div className="withdrawal-confirmation__header">
+          { rootStore.l10n.withdrawal.set_up_withdrawal_circle }
+        </div>
+        <div className="withdrawal-confirmation__content">
+          <div className="withdrawal-confirmation__form">
+            <div className="labelled-input">
+              <label htmlFor="chain">
+                { rootStore.l10n.withdrawal.fields.chain }
+              </label>
+              <Select
+                value={chain}
+                onChange={value => setChain(value)}
+                containerClassName="withdrawal-confirmation__form__select"
+                options={[
+                  ["", rootStore.l10n.withdrawal.select_chain],
+                  ...cryptoStore.CircleSupportedChains()
+                ]}
+              />
+            </div>
+            <div className="labelled-input">
+              <label htmlFor="address">
+                { rootStore.l10n.withdrawal.fields.address }
+              </label>
+              <input
+                className="labelled-input__input--address"
+                type="address"
+                value={address}
+                onChange={event => setAddress(event.target.value)}
+              />
+            </div>
+          </div>
+          {
+            errorMessage ?
+              <div className="withdrawal-confirmation__error">
+                { errorMessage }
+              </div> : null
+          }
+          <div className="withdrawal-confirmation__actions">
+            <button className="action" onClick={() => Cancel()}>
+              { rootStore.l10n.actions.cancel }
+            </button>
+            <ButtonWithLoader
+              disabled={!chain || !address}
+              onClick={async () => {
+                try {
+                  await cryptoStore.DisconnectCircle(cryptoStore.CircleAccountId());
```

Comment:
actually, when they submit a broken address this clears the old one.  
we could save the old address and re-apply it on error instead.  let me try that.

(maybe we want a disconnect button, too?)

Reply:
is this actually necessary at all? will ConnectCircle just override the existing one if it exists?

also I thought about adding a disconnect button, but there's not really much of a point. if they don't want to use the account, they'd either switch to a different one or just not use circle withdrawal at all

Reply:
if we don't disconnect, we litter the DB with entries..  but point taken, could re-write that on the backend.  immediate issue was that it was returning the oldest one instead of the new one.

ah yea agreed, no need for disconnect.